src/Service/Resolvers/SqlMutationEngine.cs

```diff
@@ -825,5 +919,37 @@ private HttpContext GetHttpContext()
         {
             return _httpContextAccessor.HttpContext!;
         }
+
+        /// <summary>
+        /// Helper method to construct a transactionscope object at Read Committed isolation level
+        /// with the TransactionScopeAsyncFlowOption option enabled
+        /// </summary>
+        /// <seealso cref="https://learn.microsoft.com/en-us/dotnet/framework/data/transactions/implementing-an-implicit-transaction-using-transaction-scope"/>
+        /// <seealso cref="https://learn.microsoft.com/en-us/dotnet/api/system.transactions.transactionscopeoption?view=net-6.0#fields" />
+        /// <seealso cref="https://learn.microsoft.com/en-us/dotnet/api/system.transactions.transactionscopeasyncflowoption?view=net-6.0#fields" />
+        /// <returns>TransactionScope object set at Read Committed isolation level</returns>
+        private static TransactionScope ConstructReadCommittedTransactionScope()
```

Comment:
>ConstructReadCommittedTransactionScope [](http://example.com/codeflow?start=40&length=38)

Is this an issue in Cosmos  or pg/my sql? will probably need issues tracking those AND documentation updates to describe the differences in possible behavior.

Reply:
This bug is applicable for all the SQL database types. The current implementation fixes the bug for all the SQL database types.

The logic is updated to choose the isolation level based on the database type. 
MsSql and PostgreSql : Read Committed.
MySql: Repeatable Read 
These are the default isolation levels for each of the database types.